src/main/scala/mesosphere/marathon/state/PathId.scala

```diff
@@ -97,10 +118,46 @@ case class PathId(path: Seq[String], absolute: Boolean = true) extends Ordered[P
   override val hashCode: Int = scala.util.hashing.MurmurHash3.productHash(this)
 }
 
+case class AbsolutePathId(path: Seq[String]) extends PathId {
+  override val absolute: Boolean = true
+  override def asAbsolutePath: AbsolutePathId = this
+  protected def toString(delimiter: String): String =
+    path.mkString("/", delimiter, "")
+}
+
+object AbsolutePathId {
+  /**
+    * Parse the string as a path, but interpret it as relative to root, if it is relative.
+    *
+    * @param path The string representation of the path to parse
+    * @return An absolute path
+    */
+  def apply(path: String): AbsolutePathId = {
+    PathId(path).canonicalPath(PathId.root)
+  }
+
+}
+
+case class RelativePathId(path: Seq[String]) extends PathId with StrictLogging {
+  override val absolute: Boolean = false
+  override def asAbsolutePath: AbsolutePathId = {
+    /* There could be some input cases in the wild where this could be hit, but in a benign way
+     * Rather than break Marathon, lets log a loud message and hopefully those cases will be addressed as we remove
+     * this method
+     */
+    val ex = new RuntimeException("An absolute path was expected where we had a relative path")
```

Comment:
I'm not sure why this is such an evil thing. I don't think any one will watch the warnings.